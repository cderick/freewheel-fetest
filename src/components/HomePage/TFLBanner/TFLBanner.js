import React from 'react'
import PropTypes from 'prop-types'
import TflInfo from './TflInfo/TflInfo'
import DisrupInfo from './DisrupInfo/DisrupInfo'
import s from './TFLBanner.module.scss'

class TFLBanner extends React.Component {
	constructor(props) {
		super(props)
		this.state = {
			transportSelect: undefined
		}
		this.handleClick = this.handleClick.bind(this)
	}

	handleClick(transport){
		this.setState({
			transportSelect: transport,
		})
	}

	render() {
		if (!this.props.tubeBanner) return null
		const { tubeMessage } = this.props.tubeBanner
		return (
			<section>
				<div className={`container-fluid ${s.containerBackground}`}>
					<div className="row h-100">
						<div className="col my-auto text-center">
							<h1 className="display-4">{tubeMessage && tubeMessage.title ? tubeMessage.title : ''}</h1>
						</div>
					</div>
				</div>
				<div className="row">
					<TflInfo tflInfo={this.props.tflInfo} handleClick={this.handleClick} />
					<DisrupInfo transportSelect={this.state.transportSelect} />
				</div>
			</section>
		)
	}
}

TFLBanner.propTypes = {
	tubeBanner: PropTypes.objectOf(PropTypes.any),
	tflInfo: PropTypes.arrayOf(PropTypes.any),
}

TFLBanner.defaultProps = {
	tubeBanner: null,
	tflInfo: null,
}

export default TFLBanner
