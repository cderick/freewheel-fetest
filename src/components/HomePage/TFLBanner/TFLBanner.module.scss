.containerBackground {
  background:
    linear-gradient(
      rgba(0, 0, 0, 0.8),
      rgba(0, 0, 0, 0.8)
    ),
    url('../../../static/underground-banner.png');
  background-position: top;
  background-repeat: no-repeat;
  background-size: cover;
  padding: 70px 0;

  h1 {
    padding: 40px 0;
    font-weight: bolder;
    font-size: 4.3rem;
    letter-spacing: 2px;
    color: white;
  }
}
