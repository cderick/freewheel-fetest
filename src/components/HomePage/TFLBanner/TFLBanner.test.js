import React from 'react'
import renderer from 'react-test-renderer'
import TFLBanner from './TFLBanner'

const minProps = {
   tubeBanner: {
      tubeMessage: {
         title: 'This is my title'
      }
   }
}

const componentNoprops = renderer.create(<TFLBanner />)
const componentWithProps = renderer.create(<TFLBanner {...minProps}  />)

describe('Test on <TFLBanner /> component', () => {
   test('should not render with no props', () => {
      const treeNoprops = componentNoprops.toJSON()
      expect(treeNoprops).toMatchSnapshot()
   })
   test('should render with min props', () => {
      const treeProps = componentWithProps.toJSON()
      expect(treeProps).toMatchSnapshot()
   })
})