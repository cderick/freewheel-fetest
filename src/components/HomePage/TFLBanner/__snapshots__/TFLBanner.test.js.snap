// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <TFLBanner /> component should not render with no props 1`] = `null`;

exports[`Test on <TFLBanner /> component should render with min props 1`] = `
<section>
  <div
    className="container-fluid containerBackground"
  >
    <div
      className="row h-100"
    >
      <div
        className="col my-auto text-center"
      >
        <h1
          className="display-4"
        >
          This is my title
        </h1>
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-5 m-5 infoService"
    />
  </div>
</section>
`;
